dipy/tracking/mesh.py

```diff
@@ -0,0 +1,176 @@
+
+    Returns
+    -------
+    triangles_idx: [s] array
+        Randomly chosen triangles_indices
+    trilin_coord: [s,3] array
+        Randomly chosen trilinear_coordinates
+
+    See Also
+    --------
+    seeds_from_surface_coordinates, random_seeds_from_mask
+
+    Examples
+    --------
+    >>> mesh_polydata = load_polydata('mesh.vtk') # fury.io
+    >>> vts = get_polydata_vertices(mesh_polydata) # fury.utils
+    >>> triangles = get_polydata_triangles(mesh_polydata) # fury.utils
+    >>> mask = np.ones([len(triangles)])
+    >>> mask[0] = 0
+    >>> normal = np.cross(vts[triangles[:, 1]] - vts[triangles[:, 0]],
+    ... vts[triangles[:, 2]] - vts[triangles[:, 0]])
+    >>> triangles_area = 0.5*np.linalg.norm(normal,axis=-1)
+    >>> nb_seeds = 100
+    >>> idx, coord = random_coordinates_from_surface(len(triangles), nb_seeds,
+    ... mask, triangles_weight=triangles_area)
+    >>> seeds_pts = seeds_from_surface_coordinates(triangles, vts, idx,
+    ... coord)
+    """
+    # Compute triangles_weight in vts_mask
+    if triangles_mask is not None:
+        if triangles_weight is None:
+            triangles_weight = triangles_mask.astype(np.float)
+        else:
+            triangles_weight *= triangles_mask.astype(np.float)
+
+    # Normalize weights to have probability (sum to 1)
+    if triangles_weight is not None:
+        triangles_weight /= triangles_weight.sum()
+
+    # Set the random seed generator
+    np.random.seed(rand_gen)
+
+    # Choose random triangles
+    triangles_idx = \
+        np.random.choice(nb_triangles, size=nb_seed, p=triangles_weight)
+
+    # Choose random trilinear coordinates
+    # http://mathworld.wolfram.com/TrianglePointPicking.html
+    trilin_coord = np.random.rand(nb_seed, 3)
+    is_upper = (trilin_coord[:, 1:].sum(axis=-1) > 1.0)
+    trilin_coord[is_upper] = 1.0 - trilin_coord[is_upper]
+    trilin_coord[:, 0] = 1.0 - (trilin_coord[:, 1] + trilin_coord[:, 2])
+
+    return triangles_idx, trilin_coord
+
+
+def seeds_from_surface_coordinates(triangles, vts_values,
+                                   triangles_idx, trilinear_coord):
+    """Compute points from triangles_indices and trilinear_coord
+
+    Parameters
+    ----------
+    triangles : [n, 3] -> m array
+        A list of triangles from a mesh
+    vts_values : [m, .] array
+        List of values to interpolates from coordinates along vertices,
+        (vertices, vertices_normal, vertices_colors ...)
+    triangles_idx : [s] array
+        Specifies which triangles should be chosen (or not)
+    trilinear_coord : [s, 3] array
+        Specifies the weight/probability of choosing each triangle
+
+    Returns
+    -------
+    pts : [s, ...] array
+        Interpolated values of vertices with triangles_idx and trilinear_coord
+
+    See Also
+    --------
+    random_coordinates_from_surface
+
+    Examples
+    --------
+    >>> mesh_polydata = load_polydata('mesh.vtk') # fury.io
```

Comment:
Same comment as above. Doctests will not recognize these functions. you need to add the `import` above as usual.

```python
>>> from fury import io, utils   # skip if not HAVE_FURY
>>> mesh_polydata = io.load_polydata('mesh.vtk')  # skip if not HAVE_FURY
etc...
```

Also, you might have to use the skip directive in both case `# skip if not HAVE_FURY`  
HAVE_FURY, should be defined above with the main  import

Reply:
I tried but doest seem to work

Reply:
Strange, I need to test locally.